src/main/kotlin/net/minecrell/pluginyml/bukkit/BukkitPlugin.kt

```diff
@@ -45,13 +46,11 @@ class BukkitPlugin : PlatformPlugin<BukkitPluginDescription>("Bukkit", "plugin.y
         description.author = description.author ?: project.findProperty("author")?.toString()
     }
 
-    override fun resolve(libraries: Configuration?, description: BukkitPluginDescription) {
-        description.libraries = (
-                (description.libraries ?: listOf()) + libraries!!
-                    .resolvedConfiguration
-                    .firstLevelModuleDependencies
-                    .map { it.module.id.toString() }
-                )
+    override fun setLibraries(libraries: ResolvedComponentResult?, description: BukkitPluginDescription) {
+        val resolved = libraries?.let {
+            it.dependencies.map { d -> (d as ResolvedDependencyResult).selected.moduleVersion?.toString() ?: error("No moduleVersion for $d") }
```

Comment:
> The elements of the returned collection are declared as DependencyResult, however the dependency instances will also implement one of the following instances:
> - ResolvedDependencyResult for dependencies which were successfully resolved.
> - UnresolvedDependencyResult for dependencies which could not be resolved for some reason.

Should you use `as? ResolvedDependencyResult` to avoid cast errors?

Reply:
Only dependencies which fail to resolve will not be a ResolvedDependencyResult. I think it makes sense for the build to fail in that case. We could implement a check with a custom error message to be more clear than a CCE if you think it's needed.

Reply:
I guess the existing error "No moduleVersion for $d" would be good enough, definitely better than the CCE :)